### PR TITLE
Introducing Webviz Layout Framework (WLF)

### PR DESCRIPTION
**Pull request by rubenthoms**
Based on https://github.com/equinor/webviz-config/issues/578, this PR introduces all React components required by the new Webviz Layout Framework. That is (not exclusively):

- `WebvizContentManager`: A context with a React reducer for storing layout information, such as which view or plugin is active
- `WebvizPluginLayoutColumn`: A component for laying out view elements vertically
- `WebvizPluginLayoutRow`: A component for laying out view elements horizontally
- `WebvizPluginLoadingIndicator`: An indicator for the loading state of entire plugins
- `WebvizPluginsWrapper`: A wrapper around all plugins on one page in a Webviz application - takes care of which plugin is active and notifies the user about it
- `WebvizPluginTour`: A new implementation for displaying information about particular components in a plugin in a stepwise tour - able to change the active view and open the settings drawer and individual settings groups
- `WebvizPluginWrapper`: A wrapper around an individual plugin - registers the plugin's information in the `WebvizContentManager`
- `WebvizSettings`: A React component for displaying the contents of a settings groups
- `WebvizSettingsDrawer`: A drawer containing a view selector, the current plugin's/view's settings groups and other actions (e.g. screenshot, full screen etc.) - can be expanded or collapsed
- `WebvizSettingsGroup`: A wrapper around a settings group - enables toggling of expansion state
- `WebvizView`: A wrapper around a view - enables downloading of view dependent files
- `WebvizViewElement`: A wrapper around an individual view element - enabling toggling full screen, taking screenshots and opening a settings dialog

Furthermore, this PR introduces:
- Pinned `dash` version to 2.4.x in GitHub workflow
- Added output of `Node` and `npm` versions as well as installed `npm` packages output to GitHub workflow
- Switched to older React (and related packages) versions in order to keep old packages working
- Adjusted `Overlay` and `ScrollArea` components

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -53,8 +53,10 @@
         "notistack": "^1.0.5",
         "react-colorscales": "^0.7.3",
         "react-draggable": "^4.4.4",
+        "react-is": "^17.0.2",
         "reactour": "^1.18.3",
```
Comment (anders-kiaer) [inline react/package.json:57]: Can `reactour` be removed now, or do we need it as long as plugin projects don't use WLF?

**Review by anders-kiaer — APPROVED** [review]
😲🥇👏 Amazing.

I had a couple of questions below. Should add a `CHANGELOG.md` before merge.

File: react/package.json [hunk 2]
```diff
@@ -92,10 +95,10 @@
         "jest": "^26.6.3",
         "mini-css-extract-plugin": "^1.3.9",
         "prop-types": "^15.7.2",
-        "react": "^17.0.2",
+        "react": "^16.14.0",
```
[the remark this answers is not in the document]
Comment (rubenthoms) [reply react/package.json:98]: Yes, we had to go back with the React version.

File: react/package.json (hunk 1 above)
Comment (rubenthoms) [reply react/package.json:57]: As long as we are keeping `WebvizPluginPlaceholder`, we have to keep `reactour`. Can be removed as soon as it is removed.

**Pull request merged**